qutebrowser/browser/commands.py

```diff
@@ -245,37 +247,60 @@ def openurl(self, url=None, implicit=False,
                       clicking on a link).
             count: The tab index to open the URL in, or None.
         """
+        force_search = False
         if url is None:
             if tab or bg or window:
-                url = config.get('general', 'default-page')
+                urls = [config.get('general', 'default-page')]
             else:
                 raise cmdexc.CommandError("No URL given, but -t/-b/-w is not "
                                           "set!")
         else:
-            try:
-                url = objreg.get('quickmark-manager').get(url)
-            except urlmarks.Error:
-                try:
-                    url = urlutils.fuzzy_url(url)
-                except urlutils.InvalidUrlError as e:
-                    # We don't use cmdexc.CommandError here as this can be
-                    # called async from edit_url
-                    message.error(self._win_id, str(e))
-                    return
-        if tab or bg or window:
-            self._open(url, tab, bg, window, not implicit)
-        else:
-            curtab = self._cntwidget(count)
-            if curtab is None:
-                if count is None:
-                    # We want to open a URL in the current tab, but none exists
-                    # yet.
-                    self._tabbed_browser.tabopen(url)
-                else:
-                    # Explicit count with a tab that doesn't exist.
-                    return
+            urllist = [u for u in url.split('\n') if u.strip()]
+            if (len(urllist) > 1 and
+                    any(not urlutils.is_url(u) and
+                        urlutils.get_path_if_valid(u, check_exists=True)
+                        is None for u in urllist)):
+                urllist = [url]
+                force_search = True
+            urls = [x for x in [self._parse_url(u, force_search=force_search)
+                                for u in urllist] if x is not None]
+        for i, cur_url in enumerate(urls):
+            if not window and i > 0:
+                tab = False
+                bg = True
+            if tab or bg or window:
+                self._open(cur_url, tab, bg, window, not implicit)
             else:
-                curtab.openurl(url)
+                curtab = self._cntwidget(count)
+                if curtab is None:
+                    if count is None:
+                        # We want to open a URL in the current tab, but none
+                        # exists yet.
+                        self._tabbed_browser.tabopen(cur_url)
```

Comment:
I removed it to comply with pylint's `too-many-branches`. Do you see a way to make the function simpler or should I just suppress it?